build-script.py

```diff
@@ -636,6 +637,114 @@ def run_xctests(toolchain, build_dir, multiroot_data_file, release, verbose):
+    )
+
+
+def build_command(args):
+    try:
+        builder = Builder(
+            toolchain=args.toolchain,
+            build_dir=realpath(args.build_dir),
+            multiroot_data_file=args.multiroot_data_file,
+            release=args.release,
+            verbose=args.verbose,
+            disable_sandbox=args.disable_sandbox,
+        )
+        # Until rdar://53881101 is implemented, we cannot request a build of multiple
+        # targets simultaneously. For now, just build one product after the other.
+        builder.build("SwiftSyntax")
+        builder.build("SwiftSyntaxParser")
+        builder.build("SwiftSyntaxBuilder")
+        builder.build("SwiftSyntaxBuilderGeneration")
+    except subprocess.CalledProcessError as e:
+        printerr("FAIL: Building SwiftSyntax failed")
+        printerr("Executing: %s" % " ".join(e.cmd))
+        printerr(e.output)
+
+
+def test_command(args):
+    try:
+        builder = Builder(
+            toolchain=args.toolchain,
+            build_dir=realpath(args.build_dir),
+            multiroot_data_file=args.multiroot_data_file,
+            release=args.release,
+            verbose=args.verbose,
+            disable_sandbox=args.disable_sandbox,
+        )
+        
+        builder.build("lit-test-helper")
+        
+        success = run_tests(
+            toolchain=args.toolchain,
+            build_dir=realpath(args.build_dir),
+            multiroot_data_file=args.multiroot_data_file,
+            release=args.release,
+            filecheck_exec=realpath(args.filecheck_exec),
+            skip_lit_tests=args.skip_lit_tests,
+            verbose=args.verbose,
+        )
+        if not success:
+            # An error message has already been printed by the failing test
+            # suite
+            sys.exit(1)
+        else:
+            print("** All tests passed **")
+    except subprocess.CalledProcessError as e:
+        printerr("FAIL: Running tests failed")
+        printerr("Executing: %s" % " ".join(e.cmd))
+        printerr(e.output)
+        sys.exit(1)
+
+def verify_source_code_command(args):
+    try:
+        verify_generated_files(args.gyb_exec, verbose=args.verbose)
+
+        verify_code_generated_files(
+            toolchain=args.toolchain,
+            build_dir=realpath(args.build_dir),
+            multiroot_data_file=args.multiroot_data_file,
+            release=args.release,
+            verbose=args.verbose,
+        )
+
+        verify_c_syntax_nodes_match()
+    except subprocess.CalledProcessError as e:
+        printerr(
+            "FAIL: Gyb-generated files committed to repository do "
+            "not match generated ones. Please re-generate the "
+            "gyb-files using the following command, open a PR to the "
+            "SwiftSyntax project and merge it alongside the main PR."
+            "$ swift-syntax/build-script.py generate-source-code --toolchain <path-to-toolchain>"
```

Comment:
Nitpick: To disambiguate how the toolchain should be passed, I like to include the final `/usr` in the path. Otherwise I’m often not sure whether `<path-to-toolchain>` should point to the `.xctoolchain` bundle or `.xctoolchain/usr`

```suggestion
            "$ swift-syntax/build-script.py generate-source-code --toolchain /path/to/toolchain.xctoolchain/usr"
```